src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1098,6 +1098,18 @@ function ($a) {
                                         ->prototype('variable')
                                         ->end()
                                     ->end()
+                                    ->arrayNode('retry_strategy')
+                                        ->addDefaultsIfNotSet()
+                                        ->validate()
+                                            ->ifTrue(function ($v) { return 'messenger.retry.exponential_retry_strategy' !== $v['service'] && (isset($v['max_retries']) || isset($v['exponential_offset'])); })
+                                            ->thenInvalid('"service" be used along with either "max_retries" or "exponential_offset".')
+                                        ->end()
+                                        ->children()
+                                            ->scalarNode('service')->defaultValue('messenger.retry.exponential_retry_strategy')->end()
+                                            ->scalarNode('max_retries')->defaultValue(3)->end()
```

Comment:
Wouldn't it make more sense for these options to be part of the `dsn`?
```
DSN=amqp://...?retry[max_retry]=3&retry[exponential_offset]=0
```

Reply:
It's not logic that ultimately ends up in the transport. It more ties to a "global retry system" that you can configure on a transport-by-transport basis.

And on a personal note, I'd like to keep things out of the DSN, unless it's really transport-specific connection config. One up-side is that this retry logic can be committed.

Reply:
Should `max_retries` be `scalarNode()` or `integerNode()`? If we change it to integer, should we add `->min(0)` too?